portfolio/src/main/webapp/script.js

```diff
@@ -63,3 +63,23 @@ function tryHarderAndGiveDaisy() {
     document.getElementById("flower-image").src = "https://webstockreview.net/images/daisy-clipart-flower-gif-2.gif";
 
 }
+
+
+
+function getHelloFromButton() {
```

Comment:
This could be written a bit more succinctly with async/await keyword

```javascript
async function onButtonClick() {
  const response = await fetch('/data');
  const text = await response.text();
  const cutieTextContainer = document.getElementById('cutie-text');
  cutieTextContainer.innerText = text;
}
``` 

`await` keyword allows you to make any 'thenable' promise-like-object appear as synchronous code, unrolling callback functions and promise chains. I feel they are much easier to follow.

`fetch` returns a promise which resolves to a `Response` object, and `Response.text()` returns a promise which resolves to a string. Both of these can be awaited to make them look more like synchronous calls.

async await:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/async_function

fetch:
https://developer.mozilla.org/en-US/docs/Web/API/WindowOrWorkerGlobalScope/fetch

response:
https://developer.mozilla.org/en-US/docs/Web/API/Response